File: src/backend/internal/model/friend.go
package model

import "time"

// Friend 好友关系模型
type Friend struct {
	ID        string    `json:"id" db:"id"`
	UserID    string    `json:"user_id" db:"user_id"`
	FriendID  string    `json:"friend_id" db:"friend_id"`
	Status    string    `json:"status" db:"status"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
	// 关联字段，用于 API 响应
	FriendName string `json:"friend_name,omitempty" db:"friend_name"`
}

File: src/frontend/src/types/friend.ts
export type FriendStatus = 'pending' | 'accepted' | 'rejected' | 'blocked';

export interface Friend {
  id: string;
  user_id: string;
  friend_id: string;
  status: FriendStatus;
  friend_name?: string;
  created_at: string;
  updated_at: string;
}

export interface FriendRequest {
  id: string;
  user_id: string;
  friend_id: string;
  status: FriendStatus;
  friend_name?: string;
  created_at: string;
  updated_at: string;
}
